src/main/java/com/mabl/integration/jenkins/MablStepDeploymentRunner.java

```diff
@@ -143,19 +143,22 @@ private void execute() throws MablPlanExecutionFailure {
             try {
 
                 // Poll until we are successful or failed - note execution service is responsible for timeout
-                ExecutionResult executionResult;
-                do {
-                    Thread.sleep(pollingIntervalMilliseconds);
-                    executionResult = client.getExecutionResults(deployment.id);
-
-                    if (executionResult == null) {
-                        // No such id - this shouldn't happen
-                        throw new MablSystemException("No deployment event found for id [%s] in mabl.", deployment.id);
-                    }
+                ExecutionResult executionResult = null;
+                for (int i=0; i<2; i++) {
+                    // Rerun this loop twice in case there are plan retries
+                    do {
+                        TimeUnit.MILLISECONDS.sleep(pollingIntervalMilliseconds);
+                        executionResult = client.getExecutionResults(deployment.id);
+
```

Comment:
Sounds good! Thank you.

Reply:
Joe - we need a second loop, because if there is a retry, then another plan run is executed and we need to start polling again. In that case, there is now one more execution that we need to wait on. I can make this easier to read if that makes sense to you.

Reply:
Are you sure? If we used a failure counter, so failing out of the test run
required two completed with failure results, wouldn't the single loop just
keep running like before, until the retry completed (successfully or
failed)?